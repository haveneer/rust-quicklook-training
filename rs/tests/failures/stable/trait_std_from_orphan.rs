struct MyType;

impl From<Vec<i32>> for MyType {
    fn from(_value: Vec<i32>) -> Self {
        todo!() // orphan rule: valide car appliqué sur MyType
    }
}

impl From<MyType> for Vec<i32> {
    fn from(_value: MyType) -> Self {
        todo!() // orphan rule: valide car From<MyType> est un type monomorphé local
    }
}

impl From<String> for Vec<i32> {
    fn from(_value: String) -> Self {
        todo!() // orphan rule: invalid car tout est non local
    }
}

fn main() {}
