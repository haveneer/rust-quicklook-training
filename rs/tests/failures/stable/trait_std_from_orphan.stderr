error[E0117]: only traits defined in the current crate can be implemented for types defined outside of the crate
  --> tests/failures/stable/trait_std_from_orphan.rs:15:1
   |
15 | impl From<String> for Vec<i32> {
   | ^^^^^------------^^^^^--------
   | |    |                |
   | |    |                `Vec` is not defined in the current crate
   | |    `String` is not defined in the current crate
   | impl doesn't use only types from inside the current crate
   |
   = note: define and implement a trait or new type instead
